### ISSUE TITLE
U2F not working on local network

### ISSUE THREAD

**Issue by eddydevink** [nextcloud/twofactor_u2f#28]
The U2F plugin is working great except for one issue. It is not working when I log on within my local network. I get the screen to tap my U2F key and my U2F key starts blinking. I tap the device and it stops blinking. Then nothing happens. I will not get another page and I don't see my browser is loading another page. 

Is this a bug or is this expected behaviour?

I use nextcloud 11.0.0 and U2F plugin version 0.1.0 with chromium 55.0.2883.75 (64-bit) and a Yubikey 4 as U2F device.

**Comment by ChristophWurst** [comment]
did you use HTTPS?

**Comment by eddydevink** [comment]
Yes, I did use https.

**Comment by eddydevink** [comment]
I tried it again with firefox with the U2F addon and that doesn't work either on my internal ip. My U2F device doesn't even get triggered using firefox.

**Comment by ChristophWurst** [comment]
please check your browser's developer console. it should log the error code

**Comment by eddydevink** [comment]
I dissabled all my addons in chromium with no effect.
In the developer console I found:
jquery-migrate.min.js?v=27fd071…:2 JQMIGRATE: Migrate is installed, version 1.4.0
challenge.js:10 sign:  Array[1]
challenge.js:14 Authenticate callback Object
challenge.js:16 U2F auth failed: 4
(anonymous) @ challenge.js:16

**Comment by ChristophWurst** [comment]
> challenge.js:16 U2F auth failed: 4

interesting: https://developers.yubico.com/U2F/Libraries/Client_error_codes.html 🤔

**Comment by ChristophWurst** [comment]
ah, did you maybe register your key with a different domain than the local one you've used to sign in?

**Comment by eddydevink** [comment]
Ah, yes I registered it using my domainname.
I used a backup code to login and registered the key using my internal ip.
Logged out and U2F is working fine using my internal IP.
So no bug, just expected behaviour.
Now I have to find a way to register more then one U2F key.
I remember some posts about that so I will dig in to that.
Thank you for you comments and very fast response.

**Comment by ChristophWurst** [comment]
Great!

> Now I have to find a way to register more then one U2F key.
> I remember some posts about that so I will dig in to that.

see https://github.com/nextcloud/twofactor_u2f/issues/15 😉

**Comment by ShapeShifter499** [comment]
So u2f is registered by domain?     I just came across this issue today and thought it was a bug. Luckily I stumbled across this by searching online. 

This really should be added as a note to the README. Something along the lines of "you will have to register a key for local domain and public domain."

Is there a reason why the U2F registered key is by domain name and not by account?  Some sort of security concern?
